Address explainer feedback from TAG (#148)
webmachinelearning/webnn: explainer.md

## What’s all this then?

> [Machine Learning](https://en.wikipedia.org/wiki/Machine_learning), and in particular its subfield [Deep Learning](https://en.wikipedia.org/wiki/Deep_learning) with its various neural network architectures, enable new compelling user experiences for web applications. Use cases range from intelligent video conferencing to privacy-enhancing and accessibility-improving features. Some common use cases that apply to a wide range of web applications include e.g.:

Reviewer: ```suggestion
[Machine Learning](https://en.wikipedia.org/wiki/Machine_learning), and in particular its subfield [Deep Learning](https://en.wikipedia.org/wiki/Deep_learning) with its various neural network architectures, enables new compelling user experiences for web applications. Use cases range from improved video conferencing to accessibility-improving features, with potential improved privacy over cloud-based solutions. Some common use cases that apply to a wide range of web applications include e.g.:
```

- Person detection
- Face detection
- Semantic segmentation
- Skeleton detection
- Style transfer
- Super resolution
- Image captioning
- Machine translation
- Noise suppression

These use cases and more are elaborated in the [use cases section of the API specification](https://webmachinelearning.github.io/webnn/#usecases).

> While some of these use cases can be implemented in a constrained manner with existing Web APIs (e.g. WebGL graphics API, as demonstrated by the [WebNN API polyfill](https://github.com/webmachinelearning/webnn-polyfill) and the first-wave JS ML frameworks), no access to platform capabilities beneficial for ML such as dedicated ML hardware accelerators constraints the scope of experiences and leads to inefficient implementations on modern hardware. This disadvantages the web platform in comparison to native platforms.

Reviewer: ```suggestion
While some of these use cases can be implemented in a constrained manner with existing Web APIs (e.g. WebGL graphics API, as demonstrated by the [WebNN API polyfill](https://github.com/webmachinelearning/webnn-polyfill) and the first-wave JS ML frameworks), the lack of access to platform capabilities beneficial for ML such as dedicated ML hardware accelerators constraints the scope of experiences and leads to inefficient implementations on modern hardware. This disadvantages the web platform in comparison to native platforms.
```

By defining the WebNN API as a model format-agnostic set of neural network operations, we shift the focus of the design towards the abstraction between the web browsers and the underlying operating system services and let the web applications and JavaScript frameworks continue to focus on satisfying the needs of the web developers knowing that the neural networks they create will faithfully execute regardless of the browser's underlying platform. What we believe works in our favor is the significant overlap of neural network operations and algorithms across all popular frameworks today. Models available in one format are generally convertible to another with little loss.

> A Model Loader API can also be built atop a neural network API without losing the appeal in its simplicity. Our view is that the two APIs are complementary and not mutually exclusive to each other; however we must start with the neural network API to ensure cross-platform interoperability, a cornerstone of the web platform.

Reviewer 2: ```suggestion
A Model Loader API can also be built atop a Web Neural Network API without losing the appeal in its simplicity. Our view is that the two APIs are complementary and not mutually exclusive to each other; however we must start with the Web Neural Network API to ensure cross-platform interoperability, a cornerstone of the web platform.
```
